tests/test_collect.py

```diff
@@ -343,3 +343,63 @@ def test_collect_version_in_config(cli_invoke, changelog_d, temp_dir):
+
+
+def test_collect_linux_newlines(cli_invoke, changelog_d, temp_dir):
+    """Verify that existing Linux newlines are preserved during collection."""
+
+    changelog = temp_dir / "CHANGELOG.rst"
+    existing_text = "Line one\nLine two"
+    with changelog.open("wb") as file:
+        file.write(existing_text.encode("utf8"))
+    (changelog_d / "20170616_nedbat.rst").write_text(COMMENT + FRAG1 + COMMENT)
+    with freezegun.freeze_time("2020-02-25T15:18:19"):
+        cli_invoke(["collect"])
+    with changelog.open("rb") as file:
+        new_text = file.read().decode("utf8")
+    windows_newline_count = new_text.count("\r\n")
+    linux_newline_count = new_text.count("\n") - windows_newline_count
+    mac_newline_count = new_text.count("\r") - windows_newline_count
+    assert linux_newline_count, "Linux newlines were not preserved"
+    assert mac_newline_count == 0, "Linux newlines were not preserved"
+    assert windows_newline_count == 0, "Linux newlines were not preserved"
+
+
+def test_collect_windows_newlines(cli_invoke, changelog_d, temp_dir):
+    """Verify that existing Windows newlines are preserved during collection."""
+
+    changelog = temp_dir / "CHANGELOG.rst"
+    existing_text = "Line one\r\nLine two"
+    with changelog.open("wb") as file:
+        file.write(existing_text.encode("utf8"))
+    (changelog_d / "20170616_nedbat.rst").write_text(COMMENT + FRAG1 + COMMENT)
+    with freezegun.freeze_time("2020-02-25T15:18:19"):
+        cli_invoke(["collect"])
+    with changelog.open("rb") as file:
+        new_text = file.read().decode("utf8")
+    windows_newline_count = new_text.count("\r\n")
+    linux_newline_count = new_text.count("\n") - windows_newline_count
+    mac_newline_count = new_text.count("\r") - windows_newline_count
+    assert windows_newline_count, "Windows newlines were not preserved"
+    assert linux_newline_count == 0, "Windows newlines were not preserved"
+    assert mac_newline_count == 0, "Windows newlines were not preserved"
+
+
+def test_collect_mac_newlines(cli_invoke, changelog_d, temp_dir):
+    """Verify that existing Mac newlines are preserved during collection."""
+
+    changelog = temp_dir / "CHANGELOG.rst"
+    existing_text = "Line one\rLine two"
+    with changelog.open("wb") as file:
+        file.write(existing_text.encode("utf8"))
+    (changelog_d / "20170616_nedbat.rst").write_text(COMMENT + FRAG1 + COMMENT)
+    with freezegun.freeze_time("2020-02-25T15:18:19"):
+        cli_invoke(["collect"])
+    with changelog.open("rb") as file:
+        new_text = file.read().decode("utf8")
+    windows_newline_count = new_text.count("\r\n")
+    linux_newline_count = new_text.count("\n") - windows_newline_count
+    mac_newline_count = new_text.count("\r") - windows_newline_count
+    assert mac_newline_count, "Mac newlines were not preserved"
+    assert linux_newline_count == 0, "Mac newlines were not preserved"
+    assert windows_newline_count == 0, "Mac newlines were not preserved"
```

Comment:
These three tests are very similar.  Did you consider using `@pytest.mark.parametrize` to write it just once?

Reply:
I did indeed. However, I couldn't think of a clean way to ensure that the newline possibilities were all checked for each style of newline ("this should have a count, that shouldn't have a count", etc). I'll update the code to use parametrization.

Reply:
This has been converted to a parametrized test. I tried to make it readable. Let me know if this meets expectations. 😀